commcare_connect/opportunity/models.py

```diff
@@ -408,15 +408,16 @@ class CompletedWork(models.Model):
     reason = models.CharField(max_length=300, null=True, blank=True)
     status_modified_date = models.DateTimeField(null=True)
 
-    def save(self, *args, **kwargs):
-        if self.pk is not None:
-            old_instance = CompletedWork.objects.get(pk=self.pk)
-            if old_instance.status != self.status:
+    def __init__(self, *args, **kwargs):
```

Comment:
@calellowitz , I've tried implementing this approach, but I'm uncertain whether it's advisable. What's your take on this?

FYI: 
This method sets default values for `status` and `status_modified_date` attributes  when the object is created. If `status` is provided in `kwargs`, it will override the default `CompletedWorkStatus.incomplete` value

The direct assignment to status here is necessary because the __setattr__ method, which handles attribute setting, might otherwise throw an error when attempting to check for the existence of the status property during initialization.

Reply:
This looks good. I think it still misses some places such as queryset update methods, but it seems fine for now.